graphql_compiler/compiler/metadata.py

```diff
@@ -24,6 +24,23 @@
 )
 
 
+FilterInfo = namedtuple(
```

Comment:
These are explicitly for supporting `EXPLAIN`, but the usefulness here seems more general. What are your (and @obi1kenobi's) thoughts on adding this information as general metadata on the `LocationInfo` class? 

This would get rid of the overlap between `RecurseInfo.depth` and `LocationInfo. recursive_scopes_depth`, and the `EXPLAIN` operation is just an interpretation of metadata we know about  each location in a query.

Reply:
I have separated FilterInfo and RecurseInfo into their own record/get calls. My original intention was to keep any external users happy as any change in the LocationInfo breaks them down. Not everybody is interested with every type of information too.

Reply:
I wasn't too worried about breaking `LocationInfo` with an additive-only change, but my preference was also for doing things separately from `LocationInfo` just to keep that namedtuple immutable. This way, it gets created once and is never altered again, without any loss of expressive power and with minimal cost to ergonomics. If this feels un-ergonomic, we could easily write a helper method that takes a location and returns the combination of (Filter/Recurse/Location)Info data in whatever form we feel like is easiest to use.

I also think this metadata has uses far beyond `EXPLAIN` -- for example, we can use this same metadata together with a stream of writes for cache invalidation, or to evaluate whether to allow certain GraphQL queries based on a metadata-driven policy, etc.

Also, note that `RecurseInfo.depth` and `LocationInfo.recursive_scopes_depth` are not the same depth -- the first is the max depth of that recursion, whereas the second is the number of distinct `@recurse` directives that the current location is within (which has bearing on how the query is scheduled and executed).